Sources/uupCore/ConnectionTester.swift

```diff
@@ -65,6 +65,8 @@ class ConnectionTester {
             loops = count
         }
 
+        var successfulTests: Int = 0
```

Comment:
swift is so odd and yet completely familiar: in places it looks exactly like golang. I am assuming `var $name:$type = $value) is for local-scoped, variable declaration (and optional initialization), only (ie, cant be used against already declared names)? 

Outside of that, is there a reason to keep number of successes or failures? I can see that you are using it [here](https://github.com/jackdcasey/uup/pull/2/files#diff-9bd940f7c794264dac37db05acdd00941f595c3d4fac02b7a303141f1a2aae52R96), but that is as a boolean check only. If that its only purpose, I would declare as `status := 0` and then assign `status = 1` on failed result, regardless if previously set. 
```
...
else { 
  status = 1 # reassigning here costs us less than checking if already set 
  print failed message
}
```

And then on function exit, where we evaluate state, we can use one-liner status. 
```
exit(status)
```

If you intend to use success count at some point, then you also need to track total count, because otherwise the count is meaningless (100 success in 120 checks is different from 100 in 1 million). From my perspective, the point of the tool is to check current state, without worrying about state itself - ie, each check runs in isolation from, and is not influenced by, the previous check and potential next check. Keeping counts means you are managing state, making this tool more geared towards historical analysis than current context, which makes it more complicated. 

State is always the enemy of scale and there are lots and lots of tools that can do historical analysis: in the following example, we use `tr` and `wc` to count the number of sentences. 
```
$ # print curl response and calculate number of sentences, which should be 4
$ curl -s http://metaphorpsum.com/paragraphs/1/4 \
> | tr . \\n \
> | tee /dev/stderr \
> | wc -l \
> | xargs echo number sentences:
A polo can hardly be considered a teeming quill without also being a lemonade
 An avid rainstorm's cell comes with it the thought that the tawie rest is a leopard
 Those dugouts are nothing more than ATMS
 Extending this logic, authors often misinterpret the aunt as a gratis station, when in actuality it feels more like a deathless use
number sentences: 4
```

We can take that a step further and see how many of those sentences contain the word "the", which is comparable to evaluating total count (number of sentences) and total success count (number of sentences that contains "the"), removing that complexity entirely from process responsible for performing the check or generating the content, etc. 
```sh
$ { curl -s http://metaphorpsum.com/paragraphs/1/4 \
> | tr . \\n \
> | tee /dev/stderr /dev/fd/3 \
> | wc -l \
> | xargs echo number sentences: >&2
> } 3>&1 \
> | grep -c -i -- the \
> | xargs echo number of sentences containing \"the\":
Few can name an inby wine that isn't a sarcoid attention
 A withdrawal sees a state as an aging man
 Some posit the dozing water to be less than worldwide
 A riteless pentagon is a tramp of the mind
number sentences: 4
number of sentences containing "the": 2
```

In a prod environment, depending on scale, I would offload and isolate instrumentation of a check as a entirely separate workload, one that can be distributed and thus run in parallel : in your example, we could write a log entry to a log file on each check and then just count success/fail instances. When we need to promote instrumentation to first-class citizen, you would use a dedicates service, like prometheus, datadog, etc. Regardless of the method, the design goal remains the same: check is responsible for checking and something else does counts. 

Sorry, I am bored.

Reply:
Those are some pretty good points! I understand the boredom, it is 2020 after all 😁

My thinking behind keeping the success as a count rather than a flag, was to allow for some more verbose output in the future. For this functionality though a boolean would have worked just as well. The state is only alive for the duration the tool is running